File: src/119. Pascal's Triangle II.py
# 难度-简单
# 杨辉三角简单dp
class Solution:
    def getRow(self, rowIndex: int) -> List[int]:
        dp = [[1] + [0] * i for i in range(rowIndex + 1)]
        for i in range(rowIndex + 1):
            dp[i][i] = 1
            for j in range(1, i):
                dp[i][j] = dp[i - 1][j - 1] + dp[i - 1][j]
        return dp[rowIndex]

File: src/forOffer/119. Pascal's Triangle II.cpp
//难度-简单
//杨辉三角简单dp
class Solution {
public:
    vector<int> getRow(int rowIndex) {
        vector<vector<int>> dp(rowIndex + 1);
        for(int i = 0; i <= rowIndex; i++) {
            dp[i].resize(i + 1);
            dp[i][0] = dp[i][i] = 1;
            for(int j = 1; j < i; j++) dp[i][j] = dp[i - 1][j - 1] + dp[i - 1][j];
        }
        return dp[rowIndex];
    }
};